js/main.js

```diff
@@ -1,12 +1,15 @@
-import { renderCalendarCells, renderWeekHeader } from './calendar.js';
-import { initModalEventListeners } from './modal.js';
-import { addEventListenersToHeaderButtons} from './header.js';
+import { renderCalendarCells, renderWeekHeader } from "./calendar.js";
+import { initModalEventListeners } from "./modal.js";
+import { addEventListenersToHeaderButtons } from "./header.js";
+import { loadEventsForCurrentWeek } from "./events.js";
+
 const currentDate = new Date();
+
 function init() {
-    renderCalendarCells();
-    initModalEventListeners();
-    renderWeekHeader(currentDate);
-    addEventListenersToHeaderButtons();
+  renderCalendarCells(() => loadEventsForCurrentWeek(currentDate));
```

Comment:
as we spoke, calendar itself could handle events loading based on the date.

```suggestion
  renderCalendarCells(currentDate);
```